connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/ProtoUtils.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.common
+
+import scala.collection.JavaConverters._
+
+import com.google.protobuf.{ByteString, Message}
+import com.google.protobuf.Descriptors.FieldDescriptor
+
+private[connect] object ProtoUtils {
+  private val format = java.text.NumberFormat.getInstance()
+  private val NUM_FIRST_BYTES = 8
+
+  def abbreviateBytes(message: Message): Message = {
+    val builder = message.toBuilder
+
+    message.getAllFields.asScala.iterator.foreach {
+      case (field: FieldDescriptor, byteString: ByteString)
```

Comment:
I think it would be great if we do the same for very long regular strings. Another example might be that someone submits a 4MB SQL query and it would be good to abbreviate that as well.

Reply:
make sense.

Reply:
done

```
In [1]: query = "SELECT" + " " * 8192 + "1"

In [2]: len(query)
Out[2]: 8199

In [3]: spark.sql(query).show()
+---+                                                                           
|  1|
+---+
|  1|
+---+
```

![image](https://user-images.githubusercontent.com/7322292/231446399-e92d3d29-1a82-4b2f-8cda-e1f66d0a2616.png)